recipe-server/client/control_new/components/recipes/Listing.js

```diff
@@ -0,0 +1,185 @@
+import { Button, Col, Input, Pagination, Row, Table } from 'antd';
+import autobind from 'autobind-decorator';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { push as pushAction, Link } from 'redux-little-router';
+
+import BooleanIcon from 'control_new/components/common/BooleanIcon';
+import LoadingOverlay from 'control_new/components/common/LoadingOverlay';
+import QueryRecipes from 'control_new/components/data/QueryRecipes';
+import ListingActionBar from 'control_new/components/recipes/ListingActionBar';
+import DataList from 'control_new/components/tables/DataList';
+import * as recipeActions from 'control_new/state/recipes/actions';
+import {
+  getRecipeListingColumns,
+  getRecipeListingCount,
+  getRecipeListingFlattenedAction,
+} from 'control_new/state/recipes/selectors';
+import {
+  getCurrentURL,
+  getQueryParameter,
+} from 'control_new/state/router/selectors';
+
+
+@connect(
+  state => ({
+    columns: getRecipeListingColumns(state),
+    count: getRecipeListingCount(state),
+    getCurrentURL: queryParams => getCurrentURL(state, queryParams),
+    ordering: getQueryParameter(state, 'ordering', '-last_updated'),
+    pageNumber: parseInt(getQueryParameter(state, 'page', 1), 10),
```

Comment:
Since I bugged Osmose about this, I'll point it out here too: this should be set into its own selector to avoid playing with the data inside of `connect`. Maybe it's worth it to build a stretchier `getQueryParameter` that does parsing for you, since we seem to run into the need of parsing ints here somewhat frequently.